api/v1alpha1/memberoperatorconfig_types.go

```diff
@@ -158,6 +158,22 @@ type WebhookConfig struct {
 	// Defines the flag that determines whether to deploy the Webhook
 	// +optional
 	Deploy *bool `json:"deploy,omitempty"`
+
+	// Defines all secrets related to webhook configuration
+	// +optional
+	Secret WebhookSecret `json:"secret,omitempty"`
```

Comment:
For `omitempty` to work with structs, the variable has to be a pointer
```suggestion
	Secret *WebhookSecret `json:"secret,omitempty"`
```

Reply:
I think that it should be fine - we use that in this way at other several places. 
When you look at ToolchainConfig, then the optional struct is used without having pointer all over the place

Reply:
Right , it's a minor thing. The only thing is that those fields will show up in the CR even when empty.

Reply:
I'm not sure about ToolchainConfig and MUR.Spec.PropagatedClaims. But I've looked at the `memberoperatorconfig` for example and you can notice the empty fields:
```
....
  spec:
    console: {}
    toolchainCluster: {}
    webhook: {}
```
the `omitempty` tag doesn't seem to work there.
Again not a big issue.

Reply:
It's not a big issue but it's a good catch. I think the pattern is for the top-level config to not use pointers but for the nested config to use pointers so that they don't show if empty. WebhookSecret is a nested config so I went ahead and changed it to use a pointer.

Reply:
That's interesting... when you get the MemberOperatorConfig, then the empty values are visible, but when you get Toolchainconfig, then the same fields for the same struct (in `members` section) are hidden. 
So it works for some CRDs, but for some it doesn't :man_shrugging:

Reply:
yeah that's weird indeed 😕